tests/math_tools/test_partial_derivatives.cpp

```diff
@@ -519,6 +587,67 @@ class PartialDerivativeTestFDM : public PartialDerivativeTest<ncells_x, ncells_y
     }
 };
 
+/**
+ * @brief A class that represents a test for partial derivatives.
+ * The test can be used with finite difference method for
+ * computing partial derivatives with boundary values.
+ */
+template <std::size_t ncells_x, std::size_t ncells_y>
+class PartialDerivativeTestFDMWithBV : public PartialDerivativeTest<ncells_x, ncells_y>
+{
+private:
+    using base_type = PartialDerivativeTest<ncells_x, ncells_y>;
+
+    using typename base_type::IdxRangeX;
+    using typename base_type::IdxRangeXY;
+    using typename base_type::IdxRangeY;
+
+public:
+    PartialDerivativeTestFDMWithBV(
+            double const xmin,
+            double const xmax,
+            double const ymin,
+            double const ymax)
+        : base_type(xmin, xmax, ymin, ymax)
+    {
+        std::vector<CoordX> point_sampling_x = build_random_non_uniform_break_points(
+                base_type::m_xmin,
+                base_type::m_xmax,
+                base_type::m_ncells_x,
+                0.2);
+        ddc::init_discrete_space<typename base_type::GridX>(point_sampling_x);
+
+        std::vector<CoordY> point_sampling_y = build_random_non_uniform_break_points(
+                base_type::m_ymin,
+                base_type::m_ymax,
+                base_type::m_ncells_y,
+                0.2);
+        ddc::init_discrete_space<typename base_type::GridY>(point_sampling_y);
+    }
+
+    template <class DerivativeDimension>
+    double compute_error(double& max_distance) const
+    {
+        IdxRangeX idxrange_x(typename base_type::IdxX(0), base_type::m_ncells_x + 1);
+        IdxRangeY idxrange_y(typename base_type::IdxY(0), base_type::m_ncells_y + 1);
+        IdxRangeXY idxrange_xy(idxrange_x, idxrange_y);
+
+        // the boundary values are the value of the test function at -pi and pi, namely 0.
+        CentralFDMPartialDerivativeWithBValueCreator<IdxRangeXY, DerivativeDimension> const
+                derivative_creator(0., 0.);
+
+        FunctionToDifferentiateLinearCosine function_to_differentiate;
+        double const max_error = base_type::template compute_max_error<
+                DerivativeDimension,
```

Comment:
Do you need to do this? Does the following not compile:
```suggestion
        double const max_error = compute_max_error<
                DerivativeDimension,
```
?
It seems simpler but it is possible that the templates mess this up